### PR TITLE
New user

### PR DESCRIPTION
**Pull request by EricBelongea**
## Description

Please include a description of what was implemented. What was changed? What was added or removed if anything? 

1. I tried to remove all coverage files and I think that this happened seeing that there are 3k red lines
2. I created a poro, facade, and service for the User. This will hit two different end points
3. Does Happy and sad path testing, but I think we need some error handling for email verification.
4. I also made a change on the backend, I had to change the user_params on the backend to not require :user becuase we are trying to create a user, therefore can't already have a user. Please enlighten me if this is so. 

Last I see there are Merge conflicts.... 

## Type of change
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Describe testing implemented and what all it covers. 

- [ ]  fully tests happy and sad paths

## Checklist

- [ ] code has been self reviewed
- [x] code runs without any errors
- [x] all tests pass

### CODE REVIEW

**Comment by EricBelongea** [comment]
I will let someone else deal with the Merge conflict, I think it's literally just the adding of '/coverage' to the .gitignore

**Comment by EricBelongea** [comment]
Also, In order to have a fully flushed out flash messages and all that I think we will need to incorporate error handling and status codes. This refers specifically to ensure that a user's email is unique. Currently the user's page breaks without a proper status/error code to push them in the right direction.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class ApplicationController < ActionController::Base
+  helper_method :current_user
+
+  def current_user
+    @_current_user ||= session[:user_id] if session[:user_id]
+  end
 end
```
Comment (ttakahashi1591) [inline app/controllers/application_controller.rb:7]: This looks good building out the sessions here for a `current_user`.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -13,4 +13,37 @@ def login_user
...
+      # else
+      #   flash[:error] = response[:error]
+      #   redirect_back(fallback_lcoation: new_user_path)
+      # end       
+    else
+      flash[:error] = "Passwords Must Match"
+      redirect_back(fallback_location: new_user_path)
+      # require 'pry'; binding.pry
+    end
+  end
```
Comment (ttakahashi1591) [inline app/controllers/users_controller.rb:43]: Do we want to just build the sad pathing before pushing this through? I am cool either way.

**Review by ttakahashi1591 — COMMENTED** [review]
This looks great and I like that you broke things down with Services, Facades and POROs.

**Pull request merged**